### PR TITLE
Add filter by topic option to search Learning Object endpoint

### PR DESCRIPTION
**Pull request by nvisal1**
***Purpose***
This PR adds support for topic filtering in the Learning Object search endpoint

***Note***
This PR requires a script to be run on the production Elasticsearch cluster to append the topic field.

### CODE REVIEW

File: src/LearningObjectSearch/drivers/LearningObjectDatastore/Elasticsearch/ElasticSearchLearningObjectDatastore/ElasticSearchLearningObjectDatastore.ts [hunk 1]
```diff
@@ -579,7 +583,12 @@ export class ElasticSearchLearningObjectDatastore
     const termsQueries: TermsQuery[] = [];
     Object.keys(filters).forEach(objectKey => {
       const termBody: { [property: string]: string[] } = {};
-      termBody[`${objectKey}.keyword`] = filters[objectKey];
+      if (objectKey === 'topic') {
```
Comment (EVogel1999) [inline src/LearningObjectSearch/drivers/LearningObjectDatastore/Elasticsearch/ElasticSearchLearningObjectDatastore/ElasticSearchLearningObjectDatastore.ts]: Could you add a quick comment explaining what is going on here?

**Comment by pzalep1** [comment]
This was converted to a draft PR 6/15/2020

File: src/LearningObjectSearch/drivers/LearningObjectDatastore/Elasticsearch/ElasticSearchLearningObjectDatastore/ElasticSearchLearningObjectDatastore.ts (hunk 1 above)
Comment (DonSeannelly) [reply src/LearningObjectSearch/drivers/LearningObjectDatastore/Elasticsearch/ElasticSearchLearningObjectDatastore/ElasticSearchLearningObjectDatastore.ts]: @EVogel1999 I did some refactoring here, can you let me know if the reasoning is clearer now?

**Comment by DonSeannelly** [comment]
I've done some refactoring and added a test around building out `TermsQuery` objects. The following is an example query from my local machine:
image

**Review by pzalep1 — COMMENTED** [review]
This looks good. Just two questions:
What is the reasoning behind only accepting one topic for the query?
What error are we returning if someone passes in more than one topic in the query? 

Would you mind updating the README for the search route with the addition of topics? If you don't have time not a big deal. I know the README is very outdated.

**Comment by DonSeannelly** [comment]
> This looks good. Just two questions:
> What is the reasoning behind only accepting one topic for the query?
> What error are we returning if someone passes in more than one topic in the query?
> 
> Would you mind updating the README for the search route with the addition of topics? If you don't have time not a big deal. I know the README is very outdated.

@pzalep1 I can't speak entirely to the reasoning, but my guess is that the planned UI had no need for showing multiple topics together. If you attempt to provide two or more topics in the query, the service will respond with a `400 Bad Request` (based on the prior implementation).

I've gone ahead and added the route to the README and documented the `topic` filter for it (trying to keep it consistent with the format of the other routes). I didn't go through and document the other query parameters that exist for the route, but they could be added to the table.

**Review by pzalep1 — APPROVED** [review]

**Review by EVogel1999 — APPROVED** [review]
Looks good to me!

**Pull request merged**